test/ssl_test.c

```diff
@@ -196,20 +196,17 @@ static int check_nid(const char *name, int expected_nid, int nid)
 
 static void print_ca_names(STACK_OF(X509_NAME) *names)
 {
-    BIO *err;
     int i;
 
     if (names == NULL || sk_X509_NAME_num(names) == 0) {
-        fprintf(stderr, "    <empty>\n");
+        TEST_note("    <empty>");
         return;
     }
-    err = BIO_new_fp(stderr, BIO_NOCLOSE);
     for (i = 0; i < sk_X509_NAME_num(names); i++) {
-        X509_NAME_print_ex(err, sk_X509_NAME_value(names, i), 4,
+        X509_NAME_print_ex(bio_err, sk_X509_NAME_value(names, i), 4,
                            XN_FLAG_ONELINE);
-        BIO_puts(err, "\n");
+        BIO_puts(bio_err, "\n");
     }
-    BIO_free(err);
```

Comment:
I started from the code in `bf_lbuf.c`, fortunately TAP indentation is a bit simpler.  At least one of the tests outputs to stdout directly but this will be changed.

This leaves the question as to the default indentation level -- match the last line with an explicit `#` or match the current test level.  Some tests add extra indentation, other outputs don't indent at all -- thus both options are going to get it wrong at times.  I'm not fussed either way, although it would be good to be consistent.  I've mostly implemented the former scheme which, I think, is the more difficult (and isn't quite working).

BTW: should the BIO implementations in `crypto/evp` and `crypto/bio` be modified to use the public function interfaces instead of directly accessing the underlying structures?  The ones in `crypto/evp` seem to use the functions, the ones in `crypto/bio` access the structures directly.  I've not checked all of them for consistency, so I'm probably wrong as to the exact demarcation.